crates/uv-resolver/src/pubgrub/dependencies.rs

```diff
@@ -159,109 +114,89 @@ impl PubGrubRequirement {
-                        url.to_string(),
-                    ));
-                };
-
                 let parsed_url = ParsedUrl::Path(ParsedPathUrl::from_source(
                     install_path.clone(),
                     lock_path.clone(),
                     url.to_url(),
                 ));
-                if !Urls::same_resource(&expected.parsed_url, &parsed_url, git) {
-                    return Err(ResolveError::ConflictingUrlsTransitive(
-                        requirement.name.clone(),
-                        expected.verbatim.verbatim().to_string(),
-                        url.verbatim().to_string(),
-                    ));
-                }
-
-                Ok(Self {
-                    package: PubGrubPackage::from_url(
-                        requirement.name.clone(),
-                        extra,
-                        requirement.marker.clone(),
-                        expected.clone(),
-                    ),
-                    version: Range::full(),
-                })
+                (url, parsed_url)
             }
             RequirementSource::Directory {
                 editable,
                 url,
                 install_path,
                 lock_path,
             } => {
-                let Some(expected) = urls.get(&requirement.name) else {
-                    return Err(ResolveError::DisallowedUrl(
-                        requirement.name.clone(),
-                        url.to_string(),
-                    ));
-                };
-
                 let parsed_url = ParsedUrl::Directory(ParsedDirectoryUrl::from_source(
                     install_path.clone(),
                     lock_path.clone(),
                     *editable,
                     url.to_url(),
                 ));
-                if !Urls::same_resource(&expected.parsed_url, &parsed_url, git) {
-                    return Err(ResolveError::ConflictingUrlsTransitive(
-                        requirement.name.clone(),
-                        expected.verbatim.verbatim().to_string(),
-                        url.verbatim().to_string(),
-                    ));
-                }
+                (url, parsed_url)
+            }
+        };
+
+        Ok(Self {
+            package: PubGrubPackage::from_package(
+                requirement.name.clone(),
+                extra,
+                requirement.marker.clone(),
+            ),
+            version: Range::full(),
+            url: Some(VerbatimParsedUrl {
+                parsed_url,
+                verbatim: verbatim_url.clone(),
+            }),
+        })
+    }
 
-                Ok(Self {
-                    package: PubGrubPackage::from_url(
-                        requirement.name.clone(),
-                        extra,
-                        requirement.marker.clone(),
-                        expected.clone(),
-                    ),
-                    version: Range::full(),
+    #[allow(clippy::too_many_arguments)]
```

Comment:
We should probably just turn this lint off. :)

Reply:
Good eye https://github.com/astral-sh/uv/pull/4529